Desktop/data/expanddataproxymodel.h

```diff
@@ -0,0 +1,54 @@
+#ifndef EXPANDDATAPROXYMODEL_H
+#define EXPANDDATAPROXYMODEL_H
+
+#include <QAbstractItemModel>
+
+class DataSetView;
+
+class ExpandDataProxyModel : public QObject
```

Comment:
Shouldnt this be a `QAbstractItemModel` or something?
Or why not?

Reply:
If it was we could reuse this for other models, although I guess that isnt to necessary.
But I think this class would also work fine if it *did* use QModelIndex

Reply:
The problem is that all the calls should not use an index, but a row and column indices.
An index returns an index of the source model: this is still used for the selectionModel, and it is used only here. I could not see how to manage to generate my own indexes that wrapped the source model indexes.